launcher/lobby/lobby_moc.cpp

```diff
@@ -358,12 +377,91 @@ void Lobby::on_connectButton_toggled(bool checked)
 	}
 	else
 	{
+		ui->connectButton->setText("Connection");
 		ui->serverEdit->setEnabled(true);
 		ui->userEdit->setEnabled(true);
+		ui->listUsers->clear();
+		hostModsMap.clear();
+		updateMods();
 		socketLobby.disconnectServer();
 	}
 }
 
+void Lobby::updateMods()
+{
+	ui->modsList->clear();
+	if(hostModsMap.empty())
+		return;
+	
+	auto enabledMods = buildModsMap();
+	for(auto & mod : hostModsMap.keys())
+	{
+		auto & modValue = hostModsMap[mod];
+		auto modName = QString("%1 (v%2)").arg(mod, modValue);
+		//first - mod name
+		//second.first - should mod be enabled
+		//second.second - is possible to resolve
+		QMap<QString, QVariant> modData;
+		QList<QVariant> modDataVal;
+		if(enabledMods.contains(mod))
+		{
+			if(enabledMods[mod] == modValue)
+				enabledMods.remove(mod); //mod fully matches, remove from list
+			else
+			{
+				modDataVal.append(true);
+				modDataVal.append(false);
+				modData[mod] = modDataVal;
+				auto * lw = new QListWidgetItem(QIcon("icons:mod-update.png"), modName); //mod version mismatch
+				lw->setData(Qt::UserRole, modData);
+				ui->modsList->addItem(lw);
+			}
+		}
+		else if(isModAvailable(mod, modValue))
+		{
+			modDataVal.append(true);
+			modDataVal.append(true);
+			modData[mod] = modDataVal;
+			auto * lw = new QListWidgetItem(QIcon("icons:mod-enabled.png"), modName); //mod available and needs to be enabled
+			lw->setData(Qt::UserRole, modData);
+			ui->modsList->addItem(lw);
+		}
+		else
+		{
+			modDataVal.append(true);
+			modDataVal.append(false);
+			modData[mod] = modDataVal;
+			auto * lw = new QListWidgetItem(QIcon("icons:mod-delete.png"), modName); //mod is not available and needs to be installed
+			lw->setData(Qt::UserRole, modData);
+			ui->modsList->addItem(lw);
+		}
+	}
+	for(auto & remainMod : enabledMods.keys())
+	{
+		//first - mod name
+		//second.first - should mod be enabled
+		//second.second - is possible to resolve
+		QMap<QString, QVariant> modData;
+		QList<QVariant> modDataVal;
+		modDataVal.append(false);
+		modDataVal.append(true);
+		modData[remainMod] = modDataVal;
```

Comment:
also impossible without temp vars?

```suggestion
		QMap<QString, QVariant> modData = {
			{ remainMod, QList<QVariant>{false, true} }
		};
```